core/build_playerglobal/src/lib.rs

```diff
@@ -62,3 +70,186 @@ pub fn build_playerglobal(
 
     Ok(())
 }
+
+// Resolve the 'name' field of a `Multiname`. This only handles the cases
+// that we need for our custom `playerglobal.swf` (
+fn resolve_multiname_name<'a>(abc: &'a AbcFile, multiname: &Multiname) -> &'a str {
+    if let Multiname::QName { name, .. } | Multiname::Multiname { name, .. } = multiname {
+        &abc.constant_pool.strings[name.0 as usize - 1]
+    } else {
+        panic!("Unexpected Multiname {:?}", multiname);
+    }
+}
+
+// Like `resolve_multiname_name`, but for namespaces instead.
+fn resolve_multiname_ns<'a>(abc: &'a AbcFile, multiname: &Multiname) -> &'a str {
+    if let Multiname::QName { namespace, .. } = multiname {
+        let ns = &abc.constant_pool.namespaces[namespace.0 as usize - 1];
+        if let Namespace::Package(p) = ns {
+            &abc.constant_pool.strings[p.0 as usize - 1]
+        } else {
+            panic!("Unexpected Namespace {:?}", ns);
+        }
+    } else {
+        panic!("Unexpected Multiname {:?}", multiname);
+    }
+}
+
+/// Handles native functons defined in our `playerglobal`
+///
+/// The high-level idea is to generate code (specifically, a `TokenStream`)
+/// which builds a table - mapping from the method ids of native functions,
+/// to Rust function pointers which implement them.
+///
+/// This table gets used when we first load a method from an ABC file.
+/// If it's a native method in our `playerglobal`, we swap it out
+/// with a `NativeMethod` retrieved from the table. To the rest of
+/// the Ruffle codebase, it appears as though the method was always defined
+/// as a naive method, and never existed in the bytecode at all.
+///
+/// See `flash.system.Security.allowDomain` for an example of defining
+/// and using a native method.
+fn write_native_table(data: &[u8], out_dir: &Path) -> Result<(), Box<dyn std::error::Error>> {
+    let mut reader = swf::avm2::read::Reader::new(data);
+    let abc = reader.read()?;
+
+    let none_tokens = quote! { None };
+    let mut rust_paths: Vec<TokenStream> = Vec::new();
+    rust_paths.resize(abc.methods.len(), none_tokens);
+
+    let mut check_trait = |trait_: &Trait, parent: Option<Index<Multiname>>| {
+        let method_id = if let TraitKind::Method { method, .. }
```

Comment:
Just for symmetry, this can be made a `match` statement with 3 cases: method/getter/setter, function, and everything else.